src/solver/optim/ortoolsImpl/linearProblem.cpp

```diff
@@ -111,21 +104,15 @@ OrtoolsMipConstraint* OrtoolsLinearProblem::addConstraint(double lb,
     }
 
     auto* mpConstraint = mpSolver_->MakeRowConstraint(lb, ub, name);
-    auto mipConstraint = std::make_unique<OrtoolsMipConstraint>(mpConstraint);
 
-    if (!mpConstraint || !mipConstraint)
+    if (!mpConstraint)
     {
         logs.error() << "Couldn't add variable to Ortools MPSolver: " << name;
     }
 
-    const auto& mapIteratorPair = constraints_.try_emplace(name, std::move(mipConstraint));
-
-    if (!mapIteratorPair.second)
-    {
-        logs.error() << "Error adding variable: " << name;
-    }
-
-    return mapIteratorPair.first->second.get(); // <<name, constraint>, bool>
+    const auto& pair = constraints_.emplace(name,
```

Comment:
can't use try_emplace because of the constructor ? if it's the case ignore this comment
same for L73